### DISCUSSION TITLE
build default extension in dockerfile

### DISCUSSION THREAD

**Discussion by huangJW7** [gitpod-io/openvscode-server#408] [Q&A]
I downloaded some extensions in advance, these extensions are suffixed with .vsix. How can I install these extensions in the dockerfile instead of manually installing them after starting the container?
[similar question](https://github.com/gitpod-io/openvscode-server/discussions/122)

I tried like this but container start failed
`USER root`

`FROM gitpod/openvscode-server:1.67.0`

`ADD vscjava.vscode-java-debug-0.43.0.vsix /home/`

`ENTRYPOINT ["/bin/sh","-c","exec ${OPENVSCODE_SERVER_ROOT}/bin/openvscode-server --host 0.0.0.0 --install-extension /home/vscjava.vscode-java-debug-0.43.0.vsix --without-connection-token \"${@}\"","--"]`

**Comment by jeanp413** [comment]
@huangJW7 you are missing the `--start-server` flag, it's needed if you are installing extensions, otherwise it will install the extensions but not start the server
